ios/browser/api/version_info/version_info_ios.mm

```diff
@@ -23,4 +37,22 @@ + (NSString*)chromiumVersion {
   return base::SysUTF8ToNSString(version_info::GetBraveChromiumVersionNumber());
 }
 
++ (NSString*)channelString {
+  return base::SysUTF8ToNSString(GetChannelString());
+}
+
++ (BraveCoreVersionInfoChannel)channel {
+  switch (GetChannel()) {
+    case version_info::Channel::STABLE:
+      return BraveCoreVersionInfoChannelStable;
+    case version_info::Channel::BETA:
+      return BraveCoreVersionInfoChannelBeta;
+    case version_info::Channel::DEV:
+      return BraveCoreVersionInfoChannelDevelopment;
+    case version_info::Channel::CANARY:
+      return BraveCoreVersionInfoChannelNightly;
+    case version_info::Channel::UNKNOWN:
+      return BraveCoreVersionInfoChannelUnknown;
+  }
+}
```

Comment:
Just FYI we can get this info at build as well: https://github.com/brave/brave-core/blob/28485a16f5955364e7f73fabf9b276eb5c064a2a/ios/BUILD.gn#L210

You'd still need to read it out of Info.plist though